src/main/java/frc/team2412/robot/subsystem/DrivebaseSubsystem.java

```diff
@@ -123,19 +127,22 @@
 
     private final PFFController<Vector2> tipController;
 
-    public DrivebaseSubsystem(SwerveModule fl, SwerveModule fr, SwerveModule bl, SwerveModule br, Gyroscope g,
-            double moduleMaxVelocityMetersPerSec) {
+    public DrivebaseSubsystem() {
+        var comp = Robot.getInstance().isCompetition();
+
         synchronized (sensorLock) {
-            gyroscope = g;
-            if (g instanceof Pigeon)
+            gyroscope = comp ? new Pigeon(GYRO_PORT) : new NavX(SerialPort.Port.kMXP);
+            if (gyroscope instanceof Pigeon)
                 gyroscope.setInverted(true);
             SmartDashboard.putData("Field", field);
         }
 
         ShuffleboardTab tab = Shuffleboard.getTab("Drivebase");
 
-        modules = new SwerveModule[] { fl, fr, bl, br };
-        this.moduleMaxVelocityMetersPerSec = moduleMaxVelocityMetersPerSec;
+        modules = new SwerveModule[] { FRONT_LEFT_CONFIG.create(comp),
+                FRONT_RIGHT_CONFIG.create(comp), BACK_LEFT_CONFIG.create(comp),
+                BACK_RIGHT_CONFIG.create(comp) };
```

Comment:
Need to change `comp` to `comp && !Robot.isSimulation()` to fix simulated pose generation (as the simulated robot has encoder offsets of '0' and non-functional absolute encoders).

Reply:
See #133